### PR TITLE
buttbots can no longer speak on the radio

### PR DESCRIPTION
**Pull request by DamianX**
:cl:
 * rscdel: Buttbots can no longer talk through radios.

### CODE REVIEW

**Comment by SkowronX** [comment]
muh buttspam

**Comment by DrSnips** [comment]
Not all heros wear capes. But who knows he might.

**Comment by Ephx1** [comment]
Doesn't this essentially make them pointless? They're only ever made to butt the station radio.

**Comment by DamianX** [comment]
I'm all for removing them if that's what you're suggesting

**Comment by SonixApache** [comment]
butt a faget

**Comment by Ephx1** [comment]
I mean, if you're going to remove their only purpose, why not?

**Comment by SonixApache** [comment]
[requires server butt]

Buttbots are an ancient meme and one of butt truly absurd things that make ss13 what it is.
Removing that butt be awful.

**Comment by DamianX** [comment]
buttbots will still work, the difference is that the griff will be limited to an area rather than to everywhere that is reached by a radio

**Comment by Kurfursten** [comment]
Perhaps as a compromise they could get the huge text that the loudspeaker gets? Would this soothe the butthurt? (Sorry, couldn't resist)

**Comment by SkowronX** [comment]
>compromise 

AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA

**Comment by DrSnips** [comment]
Stages of griff so far
- [ ] Denial
- [x] Anger
- [x] Bargaining
- [ ] Depression
- [x] Acceptance

**Comment by EdXbtkOEwGw** [comment]
Just make them show coordinates.

**Comment by Hugh-G-Rection** [comment]
If this gets merged, I will create a buttbot telecomms script and buttbot assembly instructions and post them in the thread every single day.
The butts will not be silenced.

**Comment by DrSnips** [comment]
And what would that do? Get players banned for non antag grief?

**Comment by ultraplen** [comment]
I respect that dedication @Hugh-G-Rection

**Comment by TheFriggenSawbones** [comment]
you'll die for this spaghetti-man

**Comment by Diavill** [comment]
non penso proprio

**Comment by Chazach** [comment]
I #StandWithDamian

**Comment by SysadminJeroen** [comment]
As per request from @DamianX http://ss13.moe/index.php/poll/139

**Comment by Megamaw** [comment]
you literally have to cut off someone's ass to get the resources to build one of these, don't you

i don't see why it needs to be 'nerfed'.

**Comment by DamianX** [comment]
hide megamaw posts
ignore megamaw posts
do not reply to megamaw posts

**Comment by Megamaw** [comment]
are you just going to reply to everything i comment with that now

**Comment by Hugh-G-Rection** [comment]
I don't think he can hear you now. Do you want me to relay your message?

**Comment by Durgen** [comment]
The problem with buttbots is that they are used to shit up comms by nonantags. Perhaps as a compromise they can be made to only be picked up by radios when emagged.

**Comment by SGuivenne** [comment]
Buttbots already make you valid if you start shitting up comms, and it is SoP to torture people who make them into giving over location, I don't see why this needs to be changed, it isnt very common anyway.  Plus its relatively simple to filter out buttbots from the telecoms if you aren't retarded (which everyone who works telecoms is)

**Comment by DrSnips** [comment]
>Torture it out of people

You know that doesn't ever happen and that's because it doesn't work.
Can you valid the person for it? Yes but I think this is a good change since you can utterly fuck the station comms as non antag and admins view it as an ic issue so they don't/can't do squat about the spam.
You can't filter out stuff from SBRs.

**Comment by Kammerjunk** [comment]
Hugh said something pretty important re. tcomms scripts: People *have* gotten banned for making comms useless via scripts.

**Comment by gbasood** [comment]
>The problem with buttbots is that they are used to shit up comms by nonantags. 

Okay but it's a larger issue than just this. 
Buttbots can be:
 - Impossible to find
 - Impossible to find who created them
 - Clutter up the chat log so heavily that admin ghosts can't actually be admins with ghost ears on, making it harder to do their job

Antags can still just, you know, fuck up tcomms instead. 
Please offer a better solution if you have one.

**Comment by Kammerjunk** [comment]
And if a non-antag makes the radio useless through tcomms, that's actionable
If a non-antag makes the radio useless through buttbots, grandfathering discourages slapping down on that

**Comment by Megamaw** [comment]
so, this is a bigger issue than i initially thought. (thank you, gbasood, for the informative post!)

but it's clear this is a very divisive PR.

if it's needed, here are some lighter fixes we could try before pulling out this change. i think people will be less unwilling to change if we at least try to compromise, first. 

- Remove or filter buttbot messages from admin logs
- Implement a way to locate buttbots, possibly via science, robotics consoles, or detective work. (audio plays nearby occasionally? audio might be a good balance of 'is this fun for the builder of the buttbot' and 'is this immediately noticable')
- (if not already implemented, since i don't often look at the code, esp for detective) detective coding. allow detectives to scan for dna on a bot to figure out who built it, and in the case of buttbots, whose butt it is.
- i'm not sure off the top of my head if buttbots can take input from other buttbots. have them ignore that, if they don't already.

again, these are just suggestions to get ideas started. it's clear there's some pushback from the community, but if it's causing administrative issues, this needs to be resolved.

**Comment by SGuivenne** [comment]
You can have our buttbots when you pry them from our cold dead hands.

File: code/game/machinery/bots/buttbot.dm [hunk 1]
```diff
@@ -37,16 +37,17 @@ Here it is: Buttbot.
...
+	for(var/mob/O in hearers(world.view, src))
+		to_chat(O, "<b>[src]</b> beeps, \"[message]\"")
 	return
 
 
 /obj/machinery/bot/buttbot/Hear(var/datum/speech/speech, var/rendered_speech="")
 	set waitfor = 0 //Buttbots speaking should be queued after the original speech completes
-	if(prob(buttchance) && !findtext(speech.message,"butt"))
+	var/message = speech.message // need to save this here before sleeping: the speech datum may be destroyed in the meantime
+	if(prob(buttchance) && !findtext(message,"butt"))
```
Comment (Intigracy) [inline code/game/machinery/bots/buttbot.dm:48]: Here's an alternative.
```
if(prob(buttchance) && speech.speaker != src)
	if(istype(speech.speaker, /obj/machinery/bot/buttbot))
		explode()
		return
	else
		//the rest of the proc
```

tl;dr if a buttbot attempts to add butts to a message said by another buttbot it'll explode

you can still shit up the radio with a buttbot, but you can't spam the radio with buttbot**s**

Comment (DamianX) [reply code/game/machinery/bots/buttbot.dm:48]: you can still shit up comms for everyone with 1 buttbot and N radios

Comment (Intigracy) [reply code/game/machinery/bots/buttbot.dm]: People who do that should honestly just get banned for nonatag griefing.

Comment (Intigracy) [reply code/game/machinery/bots/buttbot.dm:48]: keep radio chatter disabled for buttbots, give them their own internal radio you can enable/disable with the click that makes them fart

Comment (gbasood) [reply code/game/machinery/bots/buttbot.dm:48]: Honestly I don't think is a good solution
They'd explode pretty quickly since it's an 80% chance to butt
~~And all you have to do to trigger that is say "butt" yourself~~

Comment (Intigracy) [reply code/game/machinery/bots/buttbot.dm:48]: You missed the istype.

Comment (gbasood) [reply code/game/machinery/bots/buttbot.dm:48]: Still insane though

**Comment by NigglyWiggly** [comment]
What about fucking no
I’ve never personally had trouble with buttbot spam as an admin, and I don’t even use the special window.
If it is causing such an incredible amount of spam that it makes it hard to do admin stuff, you are perfectly able to just blow the thing up easily and no one will blame you

If we want to maintain the grandfathered in buttbot shittery thing, the best solution would be to make it equivalent to reading WGW. The problem being you’d need a way to determine who made the buttbots, and you’d need to be sure they were also the one who set up the radios and shit.

**Comment by chesse20** [comment]
buttbots are so funny dont remove them from raidio then theres no point to them -_-

**Comment by DamianX** [comment]
Server poll was close but didn't pass, closing this to try something else.

**Comment by Intigracy** [comment]
Make them unable to speak on the radio without having one attached to them.

**Comment by PJB3005** [comment]
I absolutely love having my client lock up for half a second every time somebody says something on the radio because people still cling to this dead joke.

**Comment by SonixApache** [comment]
feel free to unfuck goonchat whenever
